packages/core/src/node/request/backend-request-facade.ts

```diff
@@ -0,0 +1,46 @@
+/********************************************************************************
+ * Copyright (C) 2022 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import { CancellationToken } from 'vscode-languageserver-protocol';
+import { RequestConfiguration, RequestContext, RequestOptions, RequestService } from '@theia/request-service';
+
+@injectable()
+export class BackendRequestFacade implements RequestService {
+
+    @inject(RequestService)
+    protected readonly requestService: RequestService;
+
+    configure(config: RequestConfiguration): Promise<void> {
+        return this.requestService.configure(config);
+    }
+
+    async request(options: RequestOptions, token?: CancellationToken): Promise<RequestContext> {
+        const context = await this.requestService.request(options, token);
+        // Convert the buffer to base64 before sending it to the frontend
+        // This reduces the amount of JSON data transferred massively
+        const base64Data = Buffer.from(context.buffer).toString('base64');
```

Comment:
We actually have access to this API on the frontend for decoding, as well, although we shouldn't, in theory.

Reply:
Is the `Buffer` class actually available in the frontend? I thought it was just a stub.

Reply:
I believe it's fully functional. It works [in some extensions](https://github.com/eclipse-theia/theia-cpp-extensions/blob/d537453bdc6c08fea728c54c2f9c43d7428ebb19/packages/cpp-debug/src/browser/editable-widget/memory-editable-table-widget.tsx#L206). But what you've done above with `atob` is no doubt the correct way to handle things in the browser.